libs/Tree/Directory.php

```diff
@@ -54,10 +54,47 @@ public function getIndexPage()
             return $this->children[$index_key];
         }
 
+        /*
+          If the inherit_index flag is set, then we seek child content
+         */
+        if(
+          !empty($this->getConfig()['live']['inherit_index'])
+          && $first_page = $this->seekFirstPage()
+          ){
+          return $first_page;
+        }
+
         return null;
     }
 
     /**
+     * Seek the first available page from descendants
+     * @return Content|null
+     */
+    public function seekFirstPage(){
```

Comment:
That could work with some adjustment.  The current route path looks for specific entries by name in DauxHelper::getPage(), and falls back to getIndexPage. The result of getPage() is to error so before we get a chance to intercept the call from getFirstPage() we are already rejected.  Absorbing the seek operation into getFirstPage() would require a change of the preceding logic.  We would also have to address the call to setFirstPage() happening in the findFirstPage() method, as well as the recursive issue from calling getIndex() within getFirstPage().  

As mentioned in a previous topic, I do think that we could improve the entire generator / processor / format chain of events.  One of the improvements could be to better isolate the routing so that each generator can uniquely determine whether content is available.

So basically, move more of the routing like the DauxHelper::getFile( tree, request ) stuff into something like $generator->request( $request );